### PR TITLE
Fix dashboard attendee table columns when adding user

### LINKED ISSUE #2121: Atendee list bug when adding or removing users
## What kind of an issue is this?

- Bug report

## What is the expected behaviour?
It should look normal where every box and number is under its corresponding column. It should just look normal. Even if we remove users or add users through the dashboard, nothing should change.

screenshot from 2018-09-25 21-07-06

## What is the current behaviour?
If you remove or add a user through the dashboard menu at the attendee list it will look like the screenshot above. We have randomly two boxes, the x for removing users is all the way to the right and the text "none" is occupying the remove column. If you refresh the site it will go back to expected behaviour, its only after deleteing/adding a user

screenshot from 2018-09-25 20-00-23

## How do you reproduce this problem? 
Remove or add a user to the attendee list manually.

1. go to the attendee list
2. Add a user by writing their name OR remove a user from the list
## Other information

This might be a bug which I didn't catch when I added "year of study" to the attendee list. I'm not sure if this was an issue before, but since it hasn't been brought up I will assume this is a bug from that pull request of mine

### PR DESCRIPTION
**Pull request by tokongs**
## What kind of a pull request is this?

- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [ ] I have added tests for the code I added
- [ ] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

- [ ] I have updated the build configuration
- [ ] These changes requires changes to configuration in production 
    - [ ] I have applied the required changes in production
    - [ ] I cannot apply the required changes in production before this is deployed.

## Description of changes
The table of users will now work right away when adding a user to event. The problem was that nothing checked if it was a payment event or not and the users year of study wasn't in the response. I added the year to the response and indicated whether it is a payment event or not. The table will not insert the hasPayed checkbox in the table if the event is not a payment event.  

Resolves #2121

### CODE REVIEW

File: apps/events/dashboard/utils.py [hunk 1]
```diff
@@ -77,6 +78,11 @@ def handle_add_attendee(event, user_id):
     attendee.save()
 
     resp['message'] = '%s ble meldt på %s' % (user.get_full_name(), event)
+    if Payment.objects.filter(object_id=event.attendance_event.pk).count() != 0:
```
[the remark this answers is not in the document]
Comment (tokongs) [reply apps/events/dashboard/utils.py]: Seems like a better solution!

[the remark this answers is not in the document]
Comment (oleast) [reply apps/events/dashboard/utils.py]: I did not mean to not add more of a comment on it, sorry about that.

I have not tested the above example, but it _should_ do exactly the same. I don't know if it fits inside the line length of our settings though 🙃

[the remark this answers is not in the document]
Comment (oleast) [reply apps/events/dashboard/utils.py]: Just thought of a way to do this a bit safer. Since this is using a generic relation there can be many Payment objects related to a single `object_id`, but for different kinds of models.

We can much rather just use:
``` python
resp['is_payment_event'] = event.attendance_event.payment is not None
```

Or something similar to check if it is a payment event

[the remark this answers is not in the document]
Comment (tokongs) [reply apps/events/dashboard/utils.py]: I agree that seems safer. It is updated now.

**Comment by oleast** [comment]
*TDLR at the bottom*

Ok, so I've pulled this and tested it, and while this fixes some things, it still breaks on a lot of things.

The fix just works for adding attendees, not for wait list. It still breaks in the same way as before when removing attendees.
It may even break when there are extras for the event?

This view is just a mess because it does the same thing up to **four**(?!?!) times for different cases, and changes have not been made to all cases as features have been added.

## Solution: (?)

I have a PR that fixes the four sources problems, kinda: #2351 
If we merge that, we can add the context the front-end needs in a single place, where it can be used by all actions that need it.

Changes can be made here: https://github.com/dotkom/onlineweb4/blob/c8d91fdae79b62d1264201b1919380f8a2546270/apps/events/dashboard/utils.py#L59-L81

**TLDR**: This dashboard view is even more broken than I could have ever expected. Can we please exterminate it soon?

**Comment by tokongs** [comment]
Github didn't like my rebase creating new pull request, because apparently that works better...
